Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.h

```diff
@@ -0,0 +1,30 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class DisplayableText;
+@class OWSBubbleStrokeView;
+@class TSQuotedMessage;
+
+@interface OWSQuotedMessageView : UIView
+
+@property (nonatomic, nullable, readonly) OWSBubbleStrokeView *boundsStrokeView;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+// Only needs to be called if we're going to render this instance.
+- (void)createContents;
+
+// Measurement
+- (CGSize)sizeForMaxWidth:(CGFloat)maxWidth;
+
++ (OWSQuotedMessageView *)quotedMessageViewForConversation:(TSQuotedMessage *)quotedMessage
```

Comment:
Is this a perf optimization - to provide the displayable text when it's already available rather than generate it every time?

Reply:
Nope these are factory methods.  I've added comments to that effect.